packages/@aws-cdk/core/lib/environment-component.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * An enum-like class that represents the result of comparing either two {@link IEnvComponent}s,
+ * or a {@link IEnvComponent}, and a string.
+ */
+export class EnvComponentComparison {
+  /**
+   * This means we're certain the two components are NOT
+   * Tokens, and identical.
+   */
+  public static readonly SAME = new EnvComponentComparison();
+
+  /**
+   * This means we're certain the two components are NOT
+   * Tokens, and different.
+   */
+  public static readonly DIFFERENT = new EnvComponentComparison();
+
+  /** This means exactly one of the components is a Token. */
+  public static readonly ONE_UNRESOLVED = new EnvComponentComparison();
+
+  /** This means both components are Tokens. */
+  public static readonly BOTH_UNRESOLVED = new EnvComponentComparison();
+
+  private constructor() {
+  }
+
+  /**
+   * Returns true if:
+   *   - Neither component was a Token, and they were equal to each other.
+   * Or:
+   *   - Both component were Tokens.
+   *
+   * @returns true if `this` is either SAME or BOTH_UNRESOLVED,
+   *   false if `this` is either DIFFERENT or ONE_UNRESOLVED
+   */
+  public equalOrBothUnresolved(): boolean {
```

Comment:
I'm kinda not enjoying the very mechanistic method names here. Can we not replace these with an interpretation of the situation?

```
equalOrBothUnresolved -> equalish (to stay in ternary logic land)
equalOrAnyUnresolved -> negate and rename to definitelyDifferent
unEqual -> drop this, it's just the inverse of equalish
```

Reply:
I'm sorry, but I have no idea what `equalish` is supposed to mean.

If I see code like this:

```ts
        if (this.account.compare(policy.account).equalOrAnyUnresolved()) {
```

I roughly understand what is the condition that is being checked.

This, on the other hand:

```ts
        if (this.account.compare(policy.account).equalish()) {
```

I have no idea what that means. I have no idea looking at this code when is this `if` true, and when it is `false`.

I'm very open to having better names for these, but I strongly feel `equalish` is not it.

Reply:
Removed the helpers in favor of using them inline where they are needed.